azure/durable_functions/decorators/durable_app.py

```diff
@@ -198,6 +198,17 @@ async def df_client_middleware(*args, **kwargs):
             # Invoke user code with rich DF Client binding
             return await user_code(*args, **kwargs)
 
+        # Todo: This feels awkward - however, there are two reasons that I can't naively implement
+        # this in the same way as entities and orchestrators:
+        # 1. We intentionally wrap this exported signature with @wraps, to preserve the original
+        #    signature of the user code. This means that we can't just assign a new object to the
+        #    fb._function._func, as that would overwrite the original signature.
+        # 2. I have not yet fully tested the behavior of overriding __call__ on an object with an
+        #    async method.
+        # Here we lose type hinting and auto-documentation - not great. Need to find a better way
+        # to do this. 
```

Comment:
Did the Python team give any suggestions about how we can keep type hinting and auto-documentation?

Reply:
I didn't ask. Although their familiarity with Python generally might come in handy, getting them up to speed with this Durable code would take some effort. 
I think this is no big deal, personally. All the local unit testing I've done with Functions loses type hinting at the .build() call in this line from the Functions Python SDK, way before the Durable objects become involved. This means that in practice, users aren't going to benefit from this documentation without the Functions Python team doing refactoring in the Python SDK anyway. 
Let me know if you think differently on this